### ISSUE TITLE
Tensorflow inference slow down: Repeatedly load and delete Keras model

### ISSUE THREAD

**Issue by odieXin** [tensorflow/tensorflow#58454]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

binary

### Tensorflow Version

2.10

### Custom Code

Yes

### OS Platform and Distribution

Mac OS M1 Pro

### Python version

3.9

### Current Behaviour?

```shell
I am using Tensorflow-Macos together with Tensorflow-Metal on M1 Pro.
After loading the model and I run inference with . Everything goes well.
When I delete the model, re-build the model, re-load the weight, and re-run ```predict_on_batch```, the performance is 20% slower. What's the reason? How can I fix it?
```

### Standalone code to reproduce the issue

```shell
See descriptions above.
```

### Relevant log output

_No response_</details>

**Comment by odieXin** [comment]
This problem can be reproduced without running the first inference too.

Inference time 1:
```
Initialize Keras model
Load weight
RunInference
```

Inference time 2:
```
Initialize Keras model
Load weight
Delete weight
RunInference
```

Inference time 2 is significantly longer than inference time 1.
Any thoughts?

**Comment by bhack** [comment]
Have you tried to introduce [ tf.keras.backend.clear_session
](https://www.tensorflow.org/api_docs/python/tf/keras/backend/clear_session) before loading the weights again?

**Comment by odieXin** [comment]
Yes, @bhack I tried [tf.keras.backend.clear_session](https://www.tensorflow.org/api_docs/python/tf/keras/backend/clear_session) and it doesn't work.
I found re-loading weights doesn't impact the inference performance. 
The problem is of creating new models with Keras and the models could not be deleted clearly with the clear_session function or the del variable function in python.
gc.collect() doesn't work too.

**Comment by bhack** [comment]
We have a quite long thread at https://github.com/tensorflow/tensorflow/issues/36465

But In your specific case we are also talking about a package that is closed source and it is supported third party. Have you tried to post this case in the official forum; https://developer.apple.com/forums/tags/tensorflow-metal

**Comment by mohantym** [comment]
Hi @odieXin !
Sorry for the late response.

You can also use save the best model weights and save in checkpoints folder to start the model training again from last using [custom call back ](https://keras.io/api/callbacks/model_checkpoint/). A custom call back is recommended to stop the training when desired accuracy has been achieved to save memory usage.

That way, It will solve  the caching problem (with tf.keras.backend.clear_session())

Could you also  sample stand alone code to  replicate this issue.


Thank you
